### PR TITLE
Reviewer - deleting of comments

### PR DESCRIPTION
**Pull request by webdevelopland**
What's new:
1) **Deleting comments** functionality
2) **DiffComponent** and connected files was renamed
3) language detection
4) minor fixes

### CODE REVIEW

File: tools/reviewer/web/tslint.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
       "check-preblock"
     ],
     "typedef": [
-      true,
+      false, // false, until all typedef issues aren't fix
```
Comment (oferb) [inline tools/reviewer/web/tslint.json]: Did you mean this? s/aren't fix/are fixed
Is this a `TODO`? If so please mark it as such.

File: tools/reviewer/web/src/app/dashboard/reviews-panel/file-changes/file-changes.service.ts [hunk 2]
```diff
@@ -53,4 +57,28 @@ export class DiffService {
...
     return this.hoveredLine === i && this.isHoveredNewCode === isNewCode;
   }
+
+  // Get langulage from filename. Example:
+  // filename.js -> javascript
+  getLanguage(filename: string): string {
+    const extensionRegExp: RegExp = /(?:\.([^.]+))?$/;
+    const extension: string = extensionRegExp.exec(filename)[1];
+
+    switch (extension) {
```
Comment (oferb) [inline tools/reviewer/web/src/app/dashboard/reviews-panel/file-changes/file-changes.service.ts]: Can you add a link to the list of possible languages?

Comment (webdevelopland) [reply tools/reviewer/web/src/app/dashboard/reviews-panel/file-changes/file-changes.service.ts]: Done

File: tools/reviewer/web/tslint.json (hunk 1 above)
Comment (webdevelopland) [reply tools/reviewer/web/tslint.json]: Done

File: tools/reviewer/web/src/app/dashboard/reviews-panel/file-changes/file-changes.service.ts [hunk 3]
```diff
@@ -16,17 +16,18 @@ interface AddCommentResponse {
...
-  // Is cursor hovering above new/old code?
+  // Is cursor hovering above new code?
   private isHoveredNewCode: boolean;
 
   // Subjects, which send data:
-  lineHeightChanges: Subject<HeightResponse> = new Subject<HeightResponse>();
-  openCommentsChanges: Subject<number> = new Subject<number>();
-  closeCommentsChanges: Subject<number> = new Subject<number>();
-  newComment: Subject<AddCommentResponse> = new Subject<AddCommentResponse>();
+  lineHeightChanges = new Subject<HeightResponse>();
```
Comment (MateMahinKhan) [inline tools/reviewer/web/src/app/dashboard/reviews-panel/file-changes/file-changes.service.ts]: Are we not using strongly typed here?

Comment (webdevelopland) [reply tools/reviewer/web/src/app/dashboard/reviews-panel/file-changes/file-changes.service.ts]: We need to make decision for that situations too.
I think it's excessively with `new`.
Variable has type so you have full IDE type check functionality and it's pretty obvious for reviewer, which type it's.
I don't know, maybe we should use strongly typed way here too. But I prefer not.

**Pull request merged**